internal/services/gateway/action/invitation.go

```diff
@@ -0,0 +1,245 @@
+	cOrgInvitations, _, err := h.configstoreClient.GetUserOrgInvitations(ctx, userRef, limit)
+	if err != nil {
+		return nil, util.NewAPIError(util.KindFromRemoteError(err), err)
+	}
+
+	res := make([]*OrgInvitationResponse, len(cOrgInvitations))
+	for i, r := range cOrgInvitations {
+		org, _, err := h.configstoreClient.GetOrg(ctx, r.OrganizationID)
+		if err != nil {
+			return nil, util.NewAPIError(util.KindFromRemoteError(err), err)
+		}
+
+		res[i] = &OrgInvitationResponse{
+			OrgInvitation: r,
+			Organization:  org,
+		}
+	}
+
+	return res, nil
+}
+
+func (h *ActionHandler) GetOrgInvitations(ctx context.Context, orgRef string, limit int) ([]*cstypes.OrgInvitation, error) {
+	if !common.IsUserLogged(ctx) {
+		return nil, errors.Errorf("user not logged in")
+	}
+
+	org, err := h.GetOrg(ctx, orgRef)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get org %s:", orgRef)
+	}
+
+	isOrgOwner, err := h.IsOrgOwner(ctx, org.ID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to determine ownership")
+	}
+	if !isOrgOwner {
+		return nil, util.NewAPIError(util.ErrForbidden, errors.Errorf("user not authorized"))
+	}
+
+	orgInvitations, _, err := h.configstoreClient.GetOrgInvitations(ctx, orgRef, limit)
+	if err != nil {
+		return nil, util.NewAPIError(util.KindFromRemoteError(err), err)
+	}
+	return orgInvitations, nil
+}
+
+type CreateOrgInvitationRequest struct {
+	UserRef         string
+	OrganizationRef string
+	Role            cstypes.MemberRole
+}
+
+func (h *ActionHandler) CreateOrgInvitation(ctx context.Context, req *CreateOrgInvitationRequest) (*OrgInvitationResponse, error) {
+	if !common.IsUserLogged(ctx) {
+		return nil, errors.Errorf("user not logged in")
+	}
+
+	if h.organizationMemberAddingMode != config.OrganizationMemberAddingModeInvitation {
+		return nil, util.NewAPIError(util.ErrBadRequest, errors.Errorf("user members can not added by invitation"))
+	}
+
+	if req.UserRef == "" {
+		return nil, util.NewAPIError(util.ErrBadRequest, errors.Errorf("user id required"))
+	}
+	if req.OrganizationRef == "" {
+		return nil, util.NewAPIError(util.ErrBadRequest, errors.Errorf("organization id required"))
+	}
+	if req.Role == "" {
+		return nil, util.NewAPIError(util.ErrBadRequest, errors.Errorf("role is required"))
+	}
+
+	org, err := h.GetOrg(ctx, req.OrganizationRef)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get org %s:", req.OrganizationRef)
+	}
+
+	isOrgOwner, err := h.IsOrgOwner(ctx, org.ID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to determine ownership")
+	}
+	if !isOrgOwner {
+		return nil, util.NewAPIError(util.ErrForbidden, errors.Errorf("user not authorized"))
+	}
+
+	isOrgMember, err := h.IsOrgMember(ctx, req.UserRef, req.OrganizationRef)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to determine membership")
+	}
+	if isOrgMember {
+		return nil, util.NewAPIError(util.ErrForbidden, errors.Errorf("user is just an org member"))
+	}
+
```

Comment:
Also if already enforced in the configstore you should check here if an invitation already exists.